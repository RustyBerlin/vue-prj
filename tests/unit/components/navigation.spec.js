import { mount, RouterLinkStub, createLocalVue } from '@vue/test-utils'
import Component from '../../../components/Navigation.vue'

const $route = {
  path: '/'
}

describe('Test', () => {
  let wrapper
  const localVue = createLocalVue();
  localVue.component('router-link', RouterLinkStub);

  beforeEach(() => {
    wrapper = mount(Component, {
      mocks: {
        $route
      },
      localVue,
    })
  })

  test('snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('check nav', () => {
    expect(wrapper.contains('nav ul li')).toBe(true)
    expect(wrapper.contains('nav div.ham')).toBe(true)
  })

  test('check hamburger click', () => {
    expect(wrapper.contains('ul.hide')).not.toBe(true)
    wrapper.find('div.ham').trigger('click')
    expect(wrapper.contains('ul.hide')).toBe(true)
  })

  test('check active link', () => {
    expect(wrapper.contains('nav ul li.active')).toBe(true)
  })
})
