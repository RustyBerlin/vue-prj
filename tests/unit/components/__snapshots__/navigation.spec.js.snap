// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `"<nav><div class=\\"ham\\"></div> <ul class=\\"module\\"><li class=\\"active\\"><a>Home</a></li><li class=\\"\\"><a>About</a></li><li class=\\"\\"><a>Contact</a></li></ul></nav>"`;
